R/pre_commit.R

```diff
@@ -0,0 +1,217 @@
+#' Initialize Pre-commit Hook
+#'
+#' Creates a Git pre-commit hook that runs the `commit_hooks` function.
+#' @return The path to the pre-commit hook file if created successfully.
+#' @export
+init_hooks <- function() {
+  hook_path <- ".git/hooks/pre-commit"
+
+  # Define hook content
+  hook_content <- sprintf("#!/usr/bin/env Rscript\ndfeshiny::commit_hooks()")
+
+  # Ensure the hooks directory exists
+  if (!dir.exists(".git")) {
+    stop(".git directory not found. Are you inside a Git repository?")
+  }
+
+  # Write the hook file
+  writeLines(hook_content, hook_path)
+
+  # Make the script executable
+  Sys.chmod(hook_path, mode = "0755")
+
+  message("Pre-commit hook created at ", hook_path)
+  return(hook_path)
+}
+
+#' Commit Hooks for Data Validation, Analytics Check, and Code Styling
+#'
+#' This wrapper function executes a series of pre-commit checks, including:
+#' - Data file validation (`data_checker`)
+#' - Google Analytics key validation (`check_analytics_key`, skipped if running
+#' in shiny_template repo)
+#' - Code styling check (`style_code`)
+#'
+#' @return TRUE if all checks pass, FALSE otherwise.
+#' @export
+commit_hooks <- function() {
+  all_pass <- TRUE
+
+  message("Running data file validation...")
+  if (!data_checker()) {
+    all_pass <- FALSE
+  }
+
+  if (!file.exists("shiny_template.Rproj")) {
+    message("Running Google Analytics check...")
+    if (!check_analytics_key()) {
+      all_pass <- FALSE
+    }
+  } else {
+    message("Exectuing in shiny_template repo, Skipping Google Analytics check...")
+  }
+
+  message("Running code styling check...")
+  if (style_code()) {
+    message("Code formatting was updated. Please review changes.")
+    all_pass <- FALSE
```

Comment:
Needs a stop here to prevent the commit going through if the test doesn't pass:
`quit(save = "no", status = 1, runLast = FALSE)`